packages/react/src/hooks/utils/useInvalidateOnBlock.ts

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react'
+import { QueryKey, useQueryClient } from 'react-query'
+
+import { useBlockNumber } from '../network-status'
+
+export function useInvalidateOnBlock({
```

Comment:
Think it's useful to have a basic test that checks the query cache between new blocks?

Not easy to do with our current testing setup, but maybe at least we should add a `test.todo`.